e2e/playwright/tests/todo_plugin.spec.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// ***************************************************************
+
+import {expect, test} from '@e2e-support/test_fixture';
+import SlashCommandSuggestions from 'support/components/slash_commands';
+import {fillMessage, getTodoBotDMPageURL} from 'support/utils';
+
+export default {
+    connected: () => {
+        test.describe('available commands', () => {
+            test('with just the main command', async ({pages, page, pw}) => {
+               
+            const {adminClient, adminUser} = await pw.getAdminClient();
+            if (adminUser === null) {
+                throw new Error('can not get adminUser');
+            }
+            const dmURL = await getTodoBotDMPageURL(adminClient, '', adminUser.id);
+            await page.goto(dmURL, {waitUntil: 'load'});
+
+            const c = new pages.ChannelsPage(page);
+            const slash = new SlashCommandSuggestions(page.locator('#suggestionList'));
```

Comment:
To make it so the test can delegate the responsibility of "what DOM selector is needed for the slash command suggestions UI element" to the `SlashCommandSuggestions` class itself

An alternative to passing `page.locator('#suggestionList')` to the `SlashCommandSuggestions`

This is making use of the [Page Object Model](https://playwright.dev/docs/pom) (or POM), which decouples the DOM structure from the test code.

```suggestion
            const slash = new SlashCommandSuggestions(page));
```

Then `SlachCOmmandSuggestions` constructor becomes:

```ts
export default class SlashCommandSuggestions {
    suggestionListLocator: Location;

    constructor(readonly page: Page) {
        this.suggestionListLocator = page.locator('#suggestionList');
    }
```